main.go

```diff
@@ -12,429 +13,264 @@ import (
-	STATS_CRIT := pflag.Float64("stats-critical", 20, "Provide the minimum for Port statistics critical")
-
-	// Ports to check if the mode 'ports' is present
-	portsToCheck := pflag.IntSlice("port", []int{1, 2, 3, 4, 5, 6, 7, 8}, "Ports to check")
+	mode := pflag.StringSlice("mode", []string{"basic"}, "Output modes to enable {basic|ports|poe|all}")
+	hostName := pflag.StringP("hostname", "H", "http://192.168.112.7", "Hostname to use")
+	username := pflag.StringP("username", "u", "", "Username for authentication")
+	password := pflag.StringP("password", "p", "", "Password for authentication")
+
+	// Thresholds
+	CPU_WARN := pflag.Float64("cpu-warning", 50, "CPU usage warning threshold")
+	CPU_CRIT := pflag.Float64("cpu-critical", 90, "CPU usage critical threshold")
+	MEM_WARN := pflag.Float64("mem-warning", 50, "RAM usage warning threshold")
+	MEM_CRIT := pflag.Float64("mem-critical", 90, "RAM usage critical threshold")
+	FAN_WARN := pflag.Float64("fan-warning", 3000, "Fan speed warning threshold")
+	TEMP_WARN := pflag.Float64("temp-warning", 50, "Temperature warning threshold")
+	TEMP_CRIT := pflag.Float64("temp-critical", 70, "Temperature critical threshold")
+	STATS_WARN := pflag.Float64("stats-warning", 5, "Port stats warning threshold")
+	STATS_CRIT := pflag.Float64("stats-critical", 20, "Port stats critical threshold")
 
+	portsToCheck := pflag.IntSlice("port", []int{1, 2, 3, 4, 5, 6, 7, 8}, "Ports to check")
 	help := pflag.BoolP("help", "h", false, "Show this help")
 
-	// Parse all the flags for later usage
 	pflag.Parse()
 
-	// Display help if the -h is given, or no username / password is provided
 	if *help || *username == "" || *password == "" {
 		pflag.Usage()
 		return
 	}
 
-	*hostName += "/api/v1"
-
-	// Trying to log in
-	err := login(*username, *password)
-	if err != nil { // Error is present, display it
+	n := netgear.NewNetgear(*hostName, *username, *password)
+	if err := n.Login(); err != nil {
 		fmt.Printf("Error while trying to login: %v\n", err)
 		return
 	}
+	defer func() { _ = n.Logout() }()
 
 	if slices.Contains(*mode, "all") {
 		*mode = append(*mode, "basic", "ports", "poe")
 	}
 
-	// worstStatus is needed for 'inheriting' the worst status from the lower levels to the top
 	worstStatus := check.OK
 
-	// Now everything is fine and the token is saved in a global variable in functions.go
-	// We can proceed with checking for every mode checked and doing corresponding stuff
-
-	//Basic output
+	// Basic check
 	if slices.Contains(*mode, "basic") {
-		var data map[string]any
-		inputData := device_info()
-		err = json.Unmarshal(inputData, &data)
+		var deviceInfo netgear.DeviceInfo
+		inputData, err := n.DeviceInfo()
 		if err != nil {
-			panic(err)
+			fmt.Printf("Error retrieving device info: %v\n", err)
+			os.Exit(int(check.Unknown))
+		}
+		if err := json.Unmarshal(inputData, &deviceInfo); err != nil {
+			fmt.Printf("Failed to parse JSON: %v\n", err)
+			os.Exit(int(check.Unknown))
 		}
```

Comment:
As I [mentioned in the library](https://github.com/Icinga/check-netgear/pull/4/files#r2481869826), the caller should not do the conversion from some magic byte slice to a type, but this should happen in the library. Afterwards, this code would look way cleaner.